spec/unit/file_system_spec.rb

```diff
@@ -889,4 +889,193 @@ def increment_counter_in_multiple_processes(file, num_procs, options)
       end
     end
   end
+
+  describe '#replace_file' do
```

Comment:
Should we test what happens if a block is not passed?

Reply:
not really, the program will raise an exception about LocalJumpError